src/DynamoCore/Interfaces/IDynamoModel.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Xml;
+using Dynamo.Models;
+
+namespace Dynamo.Interfaces
+{
+    public interface IDynamoModel : INotifyPropertyChanged
+    {
+        IEnumerable<WorkspaceModel> Workspaces { get; }
+        WorkspaceModel CurrentWorkspace { get; }
+
+        event EventHandler WorkspaceCleared;
+        event DynamoModelHandler ShutdownStarted;
+        event Action CleaningUp;
+        event EventHandler<EvaluationCompletedEventArgs> EvaluationCompleted;
+        event Action<WorkspaceModel> WorkspaceAdded;
+        event Action<WorkspaceModel> WorkspaceRemoved;
+        event Action<XmlDocument> WorkspaceOpening;
+    }
```

Comment:
Let's keep our interfaces small with minimal dependencies. This makes client code easier to write. Beware "throwing" interfaces into your code because I will scrutinize them. ;)

- Consider making this interface `internal`.
- Consider using `IWorkspaceModel` (it's "immutable")
- `WorkspaceCleared` should be `Action<WorkspaceModel>`
- What the heck is `CleaningUp`?